frankenphp.go

```diff
@@ -402,39 +403,40 @@ func updateServerContext(request *http.Request, create bool, isWorkerRequest boo
 	authUser, authPassword, ok := request.BasicAuth()
 	var cAuthUser, cAuthPassword *C.char
 	if ok && authPassword != "" {
-		cAuthPassword = C.CString(authPassword)
+		cAuthPassword = thread.pinCString(authPassword)
 	}
 	if ok && authUser != "" {
-		cAuthUser = C.CString(authUser)
+		cAuthUser = thread.pinCString(authUser)
 	}
 
-	cMethod := C.CString(request.Method)
-	cQueryString := C.CString(request.URL.RawQuery)
+	cMethod := thread.pinCString(request.Method)
+	cQueryString := thread.pinCString(request.URL.RawQuery)
 	contentLengthStr := request.Header.Get("Content-Length")
 	contentLength := 0
 	if contentLengthStr != "" {
 		var err error
 		contentLength, err = strconv.Atoi(contentLengthStr)
 		if err != nil {
-			return fmt.Errorf("invalid Content-Length header: %w", err)
+			// If the content length is invalid we will just set it to 0 instead of panicking
```

Comment:
Are we sure that it's a good idea? What's the real-life use case of allowing bad Content-Length?

Reply:
Hmm you are right, maybe we should reject the request instead. Returning an error here will cause a panic in non-worker mode and cause the request to hang indefinitely in worker mode. Would it be better to just return 400 as well? (Caddy already sanitizes this, but it's better to be certain)

Reply:
I changed this to return 400 instead and added a test.